Code/utility_functions.py

```diff
@@ -7,10 +10,32 @@ def flatten(values):
             final_list.append(list_value)
     return final_list
 
+
 def change_stiffness(stiffness, part):
     if stiffness in ['stiffen', 'Stiffen', 'stiff', 'Stiff', 'stif', 'Stiff', 'STIFF']:
         motion.setStiffnesses(part, 1.0)
         return 'Stiffening'
     else:
         motion.setStiffnesses(part, 0.0)
-        return 'Loosening'
\ No newline at end of file
+        return 'Loosening'
+
+
+def centre_of_mass(posture, angle1, angle2, angle3):
+    '''Returns the centre of mass relative to the big encoder.'''
+    L1 = 1.5  # length of pendulum 1 in m
+    L2 = 0.12  # length of pendulum 2 in m
+    L3 = 0.20  # length of pendulum 3 in m
+    a1 = angle1 * np.pi/180
+    a2 = angle2 * np.pi/180
+    a3 = angle3 * np.pi/180
+    x_seat = L3 * sin(a1 + a2 + a3) + L2 * sin(a1 + a2) + L1 * sin(a1)
+    y_seat = L3 * cos(a1 + a2 + a3) + l2 * cos(a1 + a2) + L1 * cos(a1)
+    if posture == "seated":
+        x_com = x_seat + 0.00065
+        y_com = y_seat + 0.1166
+    elif posture == "extended":
+        x_com = x_seat + 0.0183
+        y_com = y_seat + 0.1494
+    else:
+        return x_seat, y_seat
```

Comment:
This function is exactly what we need, could you just change the last else to raise ValueError("Position not found") because otherwise it will silently give us the wrong values :)